Set db artifacts_dir host dynamically in vagrant.yml

andreihalici opened the pull request:
Fix for issue #144

williewheeler commented:
vagrant.yml is intended to be developer-specific. For example I think it best not to assume specific host ports since they may be unavailable on the developer's machine.

For more info, see my StackOverflow answer here:
http://stackoverflow.com/questions/13065576/override-vagrant-configuration-settings-locally-per-dev/23589414#23589414

I do like the dynamic db.artifacts_dir.host though. Can we leave vagrant.yml.sample in place, continue .gitignoring vagrant.yml, remove db.artifacts_dir.host from vagrant.yml.sample, and keep the dynamic bit in Vagrantfile?

andreihalici commented:
Hello, 

I have left vagrant.yml.sample in place, added vagrant.yml back to .gitignore, removed the db.artifacts_dir.host from vagrant.yml.sample (set it to an empty string) and added the dynamic bit in Vagrant file, please do let me know if there's anything else I need to modify. 

Regarding the original intention and my expectations as a user: when I clone a Vagrant Dev Project I'm expecting to do only two things: "git clone <repo_url>" and run "vagrant up" in the newly created directory so I can start playing with the application as soon as possible and not worry about any configuration details (ports, hosts etc.).

williewheeler commented:
Hey Andrei. It may be that the approach you describe is the appropriate approach for where we are right now. Maybe we want to prioritize removing friction over development flexibility at this point. After all if the development flexibility becomes a problem then we can always reinstate the per-developer configuration.

My suggestion though would be just to inline the config into the Vagrant file (i.e. get rid of vagrant.yml and vagrant.yml.sample), just so people aren't tempted to modify vagrant.yml, thinking it's a local config. If you want to submit another PR for that I'll merge it.
